filesystem_service/tests/utils/monitor.py

```diff
@@ -0,0 +1,90 @@
+"""
+Provides useful functions for filesystem monitor tests
+"""
+
+import collections
+from multiprocessing import Manager, Process
+import os
+from time import sleep
+
+from filesystem_service.monitor import Event, FileSystemMonitor
+
+from .consts import SUBDIR_FPATH, SUBDIR_PATH
+
+
+InotifyEvent = collections.namedtuple('_INOTIFY_EVENT',
+                                      ['wd', 'mask', 'cookie', 'len'])
+EVENTS = Manager().list()
+
+
+# pylint: disable=too-few-public-methods
+class Subscriber(object):
+    """
+    Subscriber emulator
+    """
+
+    @staticmethod
+    def update(event):
+        """
+        :param event: inotify event
+        add events to shared list variable
+        """
+
+        EVENTS.append(event)
+
+
+def get_monitor_instance(folder='/test', subscriber=None):
+    """
+    :param folder: folder to be watched
+    :param subscriber: an instance of subscriber
+    :return: an instance of FileSystemMonitor
+    """
+
+    monitor = FileSystemMonitor(folder)
+    if subscriber:
+        monitor.subscribe(subscriber)
+
+    return monitor
+
+
+def gen_event_object():
+    """
+    :return: an instance of Event
+    """
+
+    i_event = (InotifyEvent(wd=1, mask=8, cookie=0, len=16),
+               ['IN_CLOSE_WRITE', 'IN_ACCESS'], b'/tmp/onedrive',
+               b'test')
+
+    return Event(i_event)
+
+
+def start_monitor_process(monitor):
+    """
+    :param monitor: an instance of monitor
+    Starts monitor in its own process in the current thread
+    """
+
+    process = Process(target=monitor.monitor)
+    process.start()
+    sleep(0.1)
+
+    return process
+
+
+def gen_events_list(monitor):
+    """
+    :param monitor: an instance of monitor
+    Fills shared variable with events
+    """
+
+    process = start_monitor_process(monitor)
+    os.system('mkdir -p {}'.format(SUBDIR_PATH))
```

Comment:
What about copying and moving?

Reply:
Copying and moving are out of unit testing scope...

Reply:
Why?

Reply:
Copying and moving is not fundamentally different from removing in this case.

Reply:
We should go through the code statements instead of doing functional testing. Removing in its turn  is supposed to test part of a code where directory is removed from the watch list.

Reply:
So you are saying that doing `rm -rf` in a test is ok, but `cp -r` is not ok?